tests/tck/features/match/VariableLengthPattern.feature

```diff
@@ -375,3 +375,108 @@ Feature: Variable length Pattern match (m to n)
       """
     Then the result should be, in any order:
       | v.player.name |
+
+  Scenario: same src and dst for variable length pattern
+    When executing query:
+      """
+      MATCH (v)-[e:like*0..0]-(v)
+      WHERE id(v) == 'Tim Duncan'
+      RETURN count(*) AS cnt
+      """
+    Then the result should be, in any order:
+      | cnt |
+      | 1   |
+    When executing query:
+      """
+      MATCH (v)-[e:like*0..2]-(v)
+      WHERE id(v) == 'Tim Duncan'
+      RETURN count(*) AS cnt
+      """
+    Then the result should be, in any order:
+      | cnt |
+      | 5   |
+    When executing query:
+      """
+      MATCH (v)-[e:like*2..3]-(v)
+      WHERE id(v) == 'Tim Duncan'
+      RETURN count(*) AS cnt
+      """
+    Then the result should be, in any order:
+      | cnt |
+      | 48  |
+    When executing query:
+      """
+      MATCH (v)-[e:like*2..3]->(v)
+      WHERE id(v) == 'Tim Duncan'
+      RETURN count(*) AS cnt
+      """
+    Then the result should be, in any order:
+      | cnt |
+      | 4   |
+    When executing query:
+      """
+      MATCH (v)-[e:like*0..]->(v)
+      WHERE id(v) == 'Tim Duncan'
+      RETURN count(*) AS cnt
+      """
+    Then the result should be, in any order:
+      | cnt |
+      | 13  |
+
+  Scenario: variable length pattern and list expression
+    # When executing query:
+    # """
+    # MATCH (v:player{name: 'Tim Duncan'})-[e:like*0..2]-(v2)
+    # WHERE size([i in e WHERE (v)-[i]-(v2) | i])>1
```

Comment:
I don't know why not to use the variable `i` in path pattern? Is it by design? @czpmango @Shylock-Hg

Reply:
@czpmango Do process this when support edge variable of pattern expr?